nexus_constructor/json/load_from_json.py

```diff
@@ -3,14 +3,18 @@
 from PySide2.QtWidgets import QWidget
 
 from nexus_constructor.component.component_type import COMPONENT_TYPES
-from nexus_constructor.json.load_from_json_utils import _find_nx_class
+from nexus_constructor.json.load_from_json_utils import (
+    _find_nx_class,
+    _find_attribute_from_list_or_dict,
+    DEPENDS_ON_IGNORE,
+)
 from nexus_constructor.json.transformation_reader import TransformationReader
 from nexus_constructor.model.component import Component
 from nexus_constructor.model.entry import Entry
 from nexus_constructor.model.instrument import Instrument
+from nexus_constructor.model.transformation import Transformation
 from nexus_constructor.ui_utils import show_warning_dialog
 
```

Comment:
I think it would be a good idea to use a doc comment to record some assumptions that are made in this implementation.
```suggestion

"""
The current implementation makes a couple of assumptions that may not hold true for all valid JSON descriptions of valid NeXus files, but are safe if the JSON was created by the NeXus Constructor:
1. All transformations exist in NXtransformations groups inside components.
2. All depends_on paths are absolute, not relative.
"""
```